src/common/search-store.ts

```diff
@@ -19,49 +44,46 @@ export class SearchStore {
    * @param query Search query from input
    */
   @action
-  onSearch(text: string[], query = this.searchQuery) {
+  public onSearch(text: string[], query = this.searchQuery): void {
     this.searchQuery = query;
 
     if (!query) {
-      this.reset();
+      return this.reset();
+    }
+
+    this.occurrences = this.findOccurrences(text, query);
 
+    if (!this.occurrences.length) {
       return;
     }
-    this.occurrences = this.findOccurences(text, query);
-    if (!this.occurrences.length) return;
 
     // If new highlighted keyword in exact same place as previous one, then no changing in active overlay
-    if (this.occurrences[this.activeOverlayIndex] !== undefined) return;
-    this.activeOverlayIndex = this.getNextOverlay(true);
+    if (this.occurrences[this.activeOverlayIndex] === undefined) {
+      this.activeOverlayIndex = this.getNextOverlay(true);
+    }
   }
 
   /**
-   * Does searching within text array, create a list of search keyword occurences.
-   * Each keyword "occurency" is saved as index of the the line where keyword founded
-   * @param text An array of any textual data (logs, for example)
+   * Does searching within text array, create a list of search keyword occurrences.
+   * Each keyword "occurrence" is saved as index of the line where keyword was found
+   * @param lines An array of any textual data (logs, for example)
    * @param query Search query from input
-   * @returns {Array} Array of line indexes [0, 0, 14, 17, 17, 17, 20...]
+   * @returns Array of line indexes [0, 0, 14, 17, 17, 17, 20...]
    */
-  findOccurences(text: string[], query: string) {
-    if (!text) return [];
-    const occurences: number[] = [];
-
-    text.forEach((line, index) => {
-      const regex = new RegExp(this.escapeRegex(query), "gi");
-      const matches = [...line.matchAll(regex)];
-
-      matches.forEach(() => occurences.push(index));
-    });
+  private findOccurrences(lines?: string[], query?: string): number[] {
+    lines ||= [];
```

Comment:
Maybe better use default parameter?
```
private findOccurrences(lines: string[] = [], query?: string): number[] {
```

Reply:
I tried that but `lines` is sometimes `null`. I will look again to see why.